app/src/main/java/com/github/sdp_begreen/begreen/activities/MainActivity.kt

```diff
@@ -69,70 +75,54 @@ class MainActivity : AppCompatActivity() {
-     *
-     * The purpose of this method is to be called while creating the activity, to prefetch
-     * the user and the image from the database, and have them directly loaded and ready to display
-     * when opening the drawer menu
-     */
-    private fun retrieveUserWithProfilePicture() {
-        lifecycleScope.launch {
-            getConnectedUser()?.also { user ->
-                connectedUserViewModel.currentUser.value = user
-                connectedUserViewModel.currentUserProfilePicture.value = user.profilePictureMetadata?.let {
-                    FirebaseDB.getUserProfilePicture(it, user.id)
-                }
-            }
-        }
-    }
-
     /**
      * Helper function to setup the user info, description and profile picture in the drawer menu
      *
-     * If the observable value have already been initialized then simply return
-     * avoid setting observe each time we open the drawer
+     * @param imageView The view that contains the image in the drawer
+     * @param usernameTW The view that contains the username in the drawer
+     * @param descriptionTW The view that contains the description in the drawer
      */
-    private fun setupDrawerUserInfo() {
-        if (drawerInitialized) return
-
-        drawerInitialized = true
-        val imageView: ImageView = findViewById(R.id.nav_drawer_profile_picture_imageview)
+    private fun setupDrawerUserInfo(
+        imageView: ImageView, usernameTW: TextView, descriptionTW: TextView
+    ) {
         if (connectedUserViewModel.currentUser.value == null) {
-            setUpUserNameAndDescription(null) // call with null to set default values
+            setUpUserNameAndDescription(null, usernameTW, descriptionTW) // call with null to set default values
         }
         if (connectedUserViewModel.currentUserProfilePicture.value == null) {
             imageView.setImageBitmap(
                 BitmapFactory.decodeResource(resources, R.drawable.blank_profile_picture))
         }
 
-        connectedUserViewModel.currentUser.observe(this) {
-            setUpUserNameAndDescription(it)
-        }
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                launch {
+                    connectedUserViewModel.currentUser.collect {
+                        setUpUserNameAndDescription(it, usernameTW, descriptionTW)
+                    }
+                }
 
-        connectedUserViewModel.currentUserProfilePicture.observe(this) {
-            imageView.setImageBitmap(
-                it ?:
-                BitmapFactory.decodeResource(resources, R.drawable.blank_profile_picture)
-            )
+                launch {
+                    connectedUserViewModel.currentUserProfilePicture.collect {
+                        imageView.setImageBitmap(it ?:
+                        BitmapFactory.decodeResource(resources, R.drawable.blank_profile_picture))
```

Comment:
confusing indentation

Reply:
You are right, I refactored some code, but forgot to adapt the indentation, thanks for spotting it out.